src/PostgREST/Config.hs

```diff
@@ -165,77 +160,78 @@ instance JustIfMaybe a a where
 instance JustIfMaybe a (Maybe a) where
   justIfMaybe a = Just a
 
+
+
 -- | Reads and parses the config and overrides its parameters from env vars, files or db settings.
-readAppConfig :: [(Text, Text)] -> Environment -> Maybe FilePath -> Maybe Text -> Maybe B.ByteString -> IO (Either Text AppConfig)
-readAppConfig dbSettings env optPath dbUriFile secretFile = do
-  -- Now read the actual config file
+readAppConfig :: [(Text, Text)] -> Maybe FilePath -> Maybe AppConfig -> IO (Either Text AppConfig)
+readAppConfig dbSettings optPath previousConf = do
+  env <- readPGRSTEnvironment
   conf <- case optPath of
     -- Both C.ParseError and IOError are shown here
     Just cfgPath -> mapLeft show <$> (try $ C.load cfgPath :: IO (Either SomeException C.Config))
     -- if no filename provided, start with an empty map to read config from environment
     Nothing -> return $ Right M.empty
 
-  pure $ mapLeft ("Error in config: " <>) $ C.runParser parseConfig =<< conf
-
+  case C.runParser (parseConfig optPath env dbSettings) =<< conf of
+    Right parsedConfig -> do
+      confWithDbUri <-
+        case previousConf of
+          Nothing ->
+            readDbUriFile parsedConfig
+          Just prevConf ->
+            pure $ parsedConfig { configDbUri = configDbUri prevConf }
+
+      Right . decodeJWKS <$> (decodeSecret =<< readSecretFile confWithDbUri)
+    Left err ->
+      return . Left $ "Error in config: " <> err
```

Comment:
Left first? Maybe? Ehh Either? ah whatever. Left Right!

Reply:
I've cleaned up a little bit - but I think we can simplify Config a lot now that we have more optional settings and the several sources. So I'd leave it at that for this PR.